File: library/src/main/java/com/trinity/recording/service/factory/RecorderServiceFactory.java
package com.trinity.recording.service.factory;


import com.trinity.recording.RecordingImplType;
import com.trinity.recording.service.RecorderService;
import com.trinity.recording.service.impl.AudioRecordRecorderServiceImpl;

public class RecorderServiceFactory {
	private static RecorderServiceFactory instance = new RecorderServiceFactory();
	private RecorderServiceFactory(){};
	public static RecorderServiceFactory getInstance(){
		return instance;
	}
	
	public RecorderService getRecorderService(RecordingImplType recordingImplType){
		RecorderService result = null;
		switch (recordingImplType) {
		case ANDROID_PLATFORM:
			result = AudioRecordRecorderServiceImpl.getInstance();
			break;
		default:
			break;
		}
		return result;
	}
}

File: library/src/main/java/com/trinity/recording/service/factory/RecorderServiceFactory.kt
package com.trinity.recording.service.factory

import com.trinity.recording.RecordingImplType
import com.trinity.recording.service.RecorderService
import com.trinity.recording.service.impl.AudioRecordRecorderServiceImpl

class RecorderServiceFactory private constructor() {

  fun getRecorderService(recordingImplType: RecordingImplType): RecorderService? {
    var result: RecorderService? = null
    when (recordingImplType) {
      RecordingImplType.ANDROID_PLATFORM -> result = AudioRecordRecorderServiceImpl.instance
    }
    return result
  }

  companion object {
    @JvmStatic
    val instance = RecorderServiceFactory()
  }
}
